x-pack/heartbeat/monitors/browser/source/project.go

```diff
@@ -39,6 +42,16 @@ func (p *ProjectSource) Validate() error {
 }
 
 func (p *ProjectSource) Fetch() error {
+	// We only need to unzip the source exactly once
+	p.mtx.Lock()
+	defer p.mtx.Unlock()
+	if p.fetched {
+		logp.L().Debugf("browser project: re-use already unpacked source: %s", p.Workdir())
+		return nil
+	}
+
+	p.fetched = true
```

Comment:
@andrewvc, can we defer this assignment until the source is successfully extracted? If it fails for a transient error, as it is it will never retry

Reply:
I've done so, but it's unlikely to be transient as we're just unpacking a file, though I suppose a full disk could be freed (unlikely on a container however).